plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -293,19 +294,20 @@ def run(self):
                     if not (skip or self.skip_repomd_steps):
                         purge.remove_repo_duplicate_nevra(self.conduit.repo_id)
 
-                # skip to the next URL in case metadata was not found, neither it was possible to
-                # sync distribution that does not have yum repo metadata
-                if not self.metadata_found:
+                # skip to the next URL in case:
+                #  - metadata was not found
+                #  - it was not possible to sync distribution that does not have yum repo metadata
+                #  - it was not the last mirror in the list
+                if not self.metadata_found and not is_last_mirror:
```

Comment:
Before that change this condition passed and  no code after try-expect was executed (because of continue statement) and the code which checks metadata and raises exception was unreachable in case metadata  is absent.
https://github.com/pulp/pulp_rpm/pull/997/files#diff-5185b3f1e5a02c40a5d198fe779c75fbR329